pytorch_lightning/utilities/apply_func.py

```diff
@@ -35,19 +35,23 @@
     Batch = type(None)
 
 
-def to_dtype_tensor(value, dtype: torch.dtype = None, device: torch.device = None):
+def to_dtype_tensor(
+    value: Union[int, float, List[Union[int, float]]],
+    dtype: Optional[torch.dtype] = None,
+    device: Union[str, torch.device] = None,
```

Comment:
I don't understand why the default is None if we raise an exception immediately. Isn't the type hint wrong here? Either this should be optional, or we remove the None default and the check if it is None below